hazelcast/src/main/java/com/hazelcast/map/impl/MapSplitBrainHandlerService.java

```diff
@@ -63,6 +63,13 @@ protected void onStoreCollection(RecordStore recordStore) {
         DefaultRecordStore defaultRecordStore = (DefaultRecordStore) recordStore;
         defaultRecordStore.getMapDataStore().reset();
         defaultRecordStore.getIndexingObserver().onDestroy(false, true);
+
+        // Removal of old mapContainer is required not to leak old
+        // state into merged cluster. An example old state that we
+        // don't want to leak into merged cluster is index state. In
+        // merged cluster, there will be a fresh mapContainer object.
+        MapContainer mapContainer = defaultRecordStore.getMapContainer();
+        mapServiceContext.removeMapContainer(mapContainer);
```

Comment:
Should we also invoke `mapContainer.onDestroy()` if `removeMapContainer` succeeds to clean up EE-side metadata? (See similar case in `PartitionContainer#destroyMap`)

Reply:
done.